core/lib/dal/src/models/storage_event.rs

```diff
@@ -86,3 +86,65 @@ impl From<StorageL2ToL1Log> for L2ToL1Log {
         }
     }
 }
+
+#[derive(sqlx::FromRow, Debug, Clone)]
+pub struct StorageWeb3LogExt {
+    pub address: Vec<u8>,
+    pub topic1: Vec<u8>,
+    pub topic2: Vec<u8>,
+    pub topic3: Vec<u8>,
+    pub topic4: Vec<u8>,
+    pub value: Vec<u8>,
+    pub block_hash: Option<Vec<u8>>,
+    pub miniblock_number: i64,
+    pub l1_batch_number: Option<i64>,
+    pub tx_hash: Vec<u8>,
+    pub tx_index_in_block: i32,
+    pub event_index_in_block: i32,
+    pub event_index_in_tx: i32,
+    pub event_index_in_block_without_eth_transfer: i32,
+    pub event_index_in_tx_without_eth_transfer: i32,
+}
+
+impl StorageWeb3LogExt {
+    pub fn into_storage_log(self, api_eth_transfer_events: ApiEthTransferEvents) -> StorageWeb3Log {
+        let event_index_in_block = match api_eth_transfer_events {
```

Comment:
Dumb question: What happens if this conversion is called with `ApiEthTransferEvents::Disabled` for non-migrated logs? Note that the latest version of #688 [contains dedicated logic](https://github.com/matter-labs/zksync-era/pull/688/commits/393a96106bc9e9d75a4abb3f711883c522b31c79) in the DAL crate to return accurate data in this case. Not sure whether this is a concern in this case though; theoretically, we could only spin up servers with `ApiEthTransferEvents::Disabled` after the migration is complete.

Reply:
I will think about how to resolve this issue, thanks for pointing out, but generally I think we can't assign the indexes without getting logs of full miniblock, so I think that we either try to call assigning everything at that point, or to do as you said